### ISSUE TITLE
Multiselection is silently discarded

### ISSUE THREAD

**Issue by thany** [openstreetmap/iD#8553]
When having multiple features selected, that selection could be discarded at the drop of a hat. Just accidentally clicking anywhere on the map and POOF gone is my selection. Imagine spending 10 bloody minutes painstakingly selecting a bunch of buildings, only to have it go "computer says no", discarding my selection like my time is worthless.

Please don't dicard a multiselection. There is a reason people select features, and discarding that is a horrible thing to do. Discarding a selection should be done by pressing Esc or something. A very specific and conscious action that isn't done accidentally.

Actually in my case it really was a bug. I tried moving the map while having some 200 buildings selected. POOF. 10 minutes wasted.

And yes, I do have to select that many, because if I edit them, say, 20 at a time, how can I know where I've been? Which ones I've done and which I've not done yet? There's no way to know, other than to select all 200 and do the edit for all of them at once. 

But for that, I need to be able to trust the editor to respect my selection and not discard it at the (sometimes even literal) drop of a hat, without my consent or permission.

**Comment by 1ec5** [comment]
Yes, unfortunately, this is a common usability pitfall with a lot of UIs that support multiple selection. iD is using a very common affordance for multiple selection, but the map does exacerbate the problem because it’s common to drag the map during multiple selection. (If you have a trackpad that supports it, I’ve found two-finger swipe to be effective at avoiding accidental clicks when panning the map.) Some specialized desktop applications, such as image editors, add selection and deselection operations to the undo stack, which avoids this problem but can frequently get in the way when you want to undo actual edits.

> And yes, I do have to select that many, because if I edit them, say, 20 at a time, how can I know where I've been? Which ones I've done and which I've not done yet? There's no way to know, other than to select all 200 and do the edit for all of them at once.

In the Map Data panel, there’s a Highlight Changes option that adds a halo around anything you’ve edited or added to the map. It isn’t a solution to the broader usability problem you’re reporting, but hopefully you’ll find it more convenient to edit in smaller batches this way.

If you’re using [the in-development version of iD](https://ideditor.netlify.app/), you can also lasso-select the buildings you want to select, which might be more efficient than selecting the buildings one by one, depending on the situation. Drag a ring around the buildings while holding down Shift, then press <kbd>|</kbd> to select the buildings instead of their vertex nodes.

**Comment by thany** [comment]
Is it possible to put selection/deselection in the "undo list" like any other change?

Adobe Lightroom, a totally unrelated program, has this feature, and might I say for a buggy slow memory hogging program, that is one brilliant feature. It's also possible to move focus to one picture among the selected ones, without losing the whole selection. And even if, ctrl+z and it's back.

Total Commander is another one, but goes for a different approach: selections are always persistent. Always! No matter what you do, selected files stay selected (well, unless you go and process that selection in some way). You have to consciously deselect them if you want them deselected.

**Comment by tyrasd** [comment]
> selections are always persistent

this would be super hard to convey to new users.

I would say this is out of scope for iD for the moment, as it is purposefully using the most common UX pattern regarding multi-selection (which admittedly has this limitation, unfortunately), see @1ec5's comment above.
